indico/migrations/versions/20191023_1820_2496c4adc7e9_add_track_principal.py

```diff
@@ -0,0 +1,304 @@
+"""Add track principal
+
+Revision ID: 2496c4adc7e9
+Revises: eefba82b42c5
+Create Date: 2019-10-02 18:20:33.866458
+"""
+
+import bisect
+
+import sqlalchemy as sa
+from alembic import op
+from sqlalchemy.dialects import postgresql
+
+from indico.core.db.sqlalchemy import PyIntEnum
+from indico.core.db.sqlalchemy.principals import PrincipalType
+
+
+# revision identifiers, used by Alembic.
+revision = '2496c4adc7e9'
+down_revision = 'a2472148d2c5'
+branch_labels = None
+depends_on = None
+
+
+def _update_event_acl_entry(conn, user_id, event_id, permissions):
+    conn.execute('''
+        UPDATE events.principals
+        SET permissions = %s
+        WHERE user_id = %s AND event_id = %s
+    ''', [(permissions, user_id, event_id)])
+
+
+def _create_track_acl_entry(conn, user_id, track_id, permissions):
+    conn.execute('''
+        INSERT INTO events.track_principals
+        (track_id, user_id, type, read_access, full_access, permissions) VALUES
+        (%s,       %s,      %s,   false,       false,       %s)
+    ''', (track_id, user_id, PrincipalType.user.value, permissions))
+
+
+def _upgrade_permissions():
+    conn = op.get_bind()
+
+    # Create track acl entries
+
+    track_reviewers_conveners = '''
+        SELECT user_id, track_id FROM events.track_abstract_reviewers as reviewers WHERE EXISTS
```

Comment:
I don't want to nit-pick but if we uppercase SQL keywords then we should also use `AS` instead of `as`